packages/providers/src.ts/base-provider.ts

```diff
@@ -1204,22 +1205,23 @@ export class BaseProvider extends Provider implements EnsProvider {
         }
     }
 
-    async getBalance(addressOrName: string | Promise<string>, blockTag?: BlockTag | Promise<BlockTag>): Promise<BigNumber> {
+    /**
+     *  AccountBalance query implementation, using the hashgraph sdk.
+     *  It returns the tinybar balance of the given address.
+     *
+     * @param addressOrName The address to check balance of
+     */
+    async getBalance(addressOrName: string | Promise<string>): Promise<BigNumber> {
         await this.getNetwork();
-        const params = await resolveProperties({
-            address: this._getAddress(addressOrName),
-            blockTag: this._getBlockTag(blockTag)
-        });
-
-        const result = await this.perform("getBalance", params);
-        try {
-            return BigNumber.from(result);
-        } catch (error) {
-            return logger.throwError("bad result from backend", Logger.errors.SERVER_ERROR, {
```

Comment:
My thinking was to try catch the actual query and not the parsing of the balance

Reply:
We should surround with try/catch only the query meaning:
```
const balance = await new AccountBalanceQuery()
                .setAccountId(new AccountId({ shard: shardNum, realm: realmNum, num: accountNum }))
                .execute(this.hederaClient);
```

and not the whole method. If something is wrong with `getAccountFromAddress`, we should not wrap the meaningful error and throw "bad result from backend" since we have not called the backend at all